[ci] SimpleQueryStringBuilderTests.testToQuery

Not sure what the right behavior is here, maybe it thinks the query shouldn't be cacheable because it's interpreting `"query" : "nOw"` as "now"?

Reproduces locally

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.4+periodic/97/console

[build-97.txt.zip](https://github.com/elastic/elasticsearch/files/2540144/build-97.txt.zip)

```
./gradlew :server:test -Dtests.seed=E44BD1DAE92D549C -Dtests.class=org.elasticsearch.index.query.SimpleQueryStringBuilderTests -Dtests.method="testToQuery" 
-Dtests.security.manager=true -Dtests.locale=he -Dtests.timezone=America/Campo_Grande
```

```
FAILURE 0.10s | SimpleQueryStringBuilderTests.testToQuery <<< FAILURES!                 
   > Throwable #1: java.lang.AssertionError: query was marked as not cacheable in the context but this test indicates it should be cacheable: {                                  
   >   "simple_query_string" : {                                                        
   >     "query" : "nOw",                                                               
   >     "analyzer" : "standard",                                                       
   >     "flags" : -1,                                                                  
   >     "default_operator" : "or",                                                     
   >     "analyze_wildcard" : false,
   >     "quote_field_suffix" : "yxzqzti",
   >     "minimum_should_match" : "75%",
   >     "auto_generate_synonyms_phrase_query" : true,
   >     "fuzzy_prefix_length" : 0,
   >     "fuzzy_max_expansions" : 50,
   >     "fuzzy_transpositions" : true,
   >     "boost" : 1.0
   >   }
   > }
   >    at __randomizedtesting.SeedInfo.seed([E44BD1DAE92D549C:13B0D3E498AE9176]:0)
   >    at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:423)
   >    at java.lang.Thread.run(Thread.java:748)
```

--- a/server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java
+++ b/server/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTests.java
@@ -104,6 +104,12 @@ public class SimpleQueryStringBuilderTests extends AbstractQueryTestCase<SimpleQ
                 fields.put(STRING_FIELD_NAME_2, 2.0f / randomIntBetween(1, 20));
             }
         }
+        // special handling if query is "now" and no field specified. This hits the "mapped_date" field which leads to the query not being
+        // cacheable and trigger later test failures (see https://github.com/elastic/elasticsearch/issues/35183)
+        if (fieldCount == 0 && result.value().equalsIgnoreCase("now")) {
+            fields.put(STRING_FIELD_NAME_2, 2.0f / randomIntBetween(1, 20));
+        }
+
         result.fields(fields);
         if (randomBoolean()) {
             result.autoGenerateSynonymsPhraseQuery(randomBoolean());

